utils.groovy

```diff
@@ -939,13 +939,44 @@ def build_and_push_image(params = [:]) {
     def manifest_digest = ""
     def extra_build_args = params.get('extra_build_args', "");
     def v2s2 = params.get('v2s2', false);
+    def arches = params['arches']
 
-    def digests = build_remote_image(params['arches'], params['src_commit'], params['src_url'], params['staging_repository'],
+    def from_manifest = skopeo_inspect(from)
+    if (!from.contains('@')) {
+        // pin the FROM with the manifest digest, to avoid arch remote builders reusing stale images
+        from = "${from.split(':')[0]}@${from_manifest["Digest"]}"
+    }
+    // also save the base image buildid
+    def base_build_id = from_manifest["Labels"]["org.opencontainers.image.version"]
+
+    def digests = build_remote_image(arches, params['src_commit'], params['src_url'], params['staging_repository'],
                                      params['image_tag_staging'], secret, from, extra_build_args)
+
     stage("Push Manifest") {
         manifest_digest = push_manifest(digests, params['staging_repository'], params['manifest_tag_staging'], v2s2)
     }
+
+    // sanity check: all the images should have the same base image
+    for (arch in arches) {
+        def manifest = skopeo_inspect("${params['staging_repository']}@${manifest_digest}", arch=arch)
+        def id = manifest["Labels"]["org.opencontainers.image.version"]
+        if (id != base_build_id) {
+            throw new Exception("Mismatched base image build id for ${arch}. Expected ${base_build_id}, got ${id}")
+        }
+    }
     return manifest_digest
 }
 
+def skopeo_inspect(image, arch = "") {
```

Comment:
How about we return everything we need here? and  do something like: 

```suggestion 
def getBaseImageInfo(buildFrom, arch "") {
   ...
   
    def manifestJson = shwrapCapture("""
        skopeo inspect --authfile ${registryAuthFile} --no-tags ${arch_arg}  docker://${image}
    """)
    ...
    
    def manifest = readJSON(text: manifestJson)
    def digest = manifest["Digest"]
    def buildId = manifest["Labels"]["org.opencontainers.image.version"]

    return [digest: digest, buildId: buildId]
} 
```

In this way we do not need to look twice for ```manifest["Labels"]["org.opencontainers.image.version"]``` for example.

Reply:
the only thing this really allows to refactor is `["Labels"]["org.opencontainers.image.version"]` at the cost of the helper not being generic anymore